crates/sovereign-sdk/module-system/sov-modules-stf-blueprint/src/stf_blueprint.rs

```diff
@@ -238,42 +232,12 @@ where
         )
     }
 
-    #[cfg_attr(all(target_os = "zkvm", feature = "bench"), cycle_tracker)]
-    pub(crate) fn _apply_soft_confirmation_inner(
-        &self,
-        checkpoint: StateCheckpoint<C>,
-        soft_confirmation: &mut SignedSoftConfirmationBatch,
-        pre_state_root: &<C::Storage as Storage>::Root,
-        current_spec: SpecId,
-    ) -> (ApplySoftConfirmationResult<Da>, StateCheckpoint<C>) {
-        let batch_workspace = checkpoint.to_revertable();
-
-        match self.begin_soft_confirmation_inner(
-            batch_workspace,
-            soft_confirmation,
-            pre_state_root,
-            current_spec,
-        ) {
-            (Ok(()), batch_workspace) => {
-                // TODO: wait for txs here, apply_sov_txs can be called multiple times
-                let (batch_workspace, tx_receipts) = self.apply_sov_txs_inner(
-                    soft_confirmation.txs(),
-                    current_spec,
-                    batch_workspace,
-                );
-
-                self.end_soft_confirmation_inner(soft_confirmation, tx_receipts, batch_workspace)
-            }
-            (Err(err), batch_workspace) => (Err(err), batch_workspace.revert()),
-        }
-    }
-
     // Stateless verification of transaction, such as signature check
     // Single malformed transaction results in sequencer slashing.
-    fn verify_txs_stateless_soft(&self, txs: &[Vec<u8>]) -> Vec<TransactionAndRawHash<C>> {
+    fn verify_txs_stateless_soft(&self, txs: Vec<Vec<u8>>) -> Vec<TransactionAndRawHash<C>> {
         verify_txs_stateless(
-            txs.iter()
-                .map(|tx| RawTx { data: tx.clone() })
+            txs.into_iter()
+                .map(|tx| RawTx { data: tx })
                 .collect::<Vec<_>>(),
         )
```

Comment:
Maybe get rid of Vec<Vec<>> and .collect()

Reply:
i don't understan